homeassistant/components/verisure/__init__.py

```diff
@@ -41,6 +43,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data.setdefault(DOMAIN, {})
     hass.data[DOMAIN][entry.entry_id] = coordinator
 
+    # Migrate lock default code from config entry to lock entity
+    if config_entry_default_code := entry.options.get(CONF_LOCK_DEFAULT_CODE):
+        entity_reg = er.async_get(hass)
+        entries = er.async_entries_for_config_entry(entity_reg, entry.entry_id)
+        for entity in entries:
+            if entity.entity_id.startswith("lock"):
+                entity_reg.async_update_entity_options(
+                    entity.entity_id,
+                    LOCK_DOMAIN,
+                    {CONF_DEFAULT_CODE: config_entry_default_code},
+                )
+        new_options = entry.options.copy()
+        del new_options[CONF_LOCK_DEFAULT_CODE]
+        hass.config_entries.async_update_entry(entry, options=new_options)
```

Comment:
I discussed this with @frenck and we think we should have a proper [config entry migration](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#config-entry-migration) here.

We don't support downgrades.
However, suppose a user downgrades with a proper config entry migration. In that case, HA will inform the user that the config entry is not compatible as the config entry version is not matching.
With the code above, the user will probably get some exceptions as you are deleting the `CONF_LOCK_DEFAULT_CODE` option.

Reply:
The first line checks if the key exist or not so there won't be any exception.
Don't think a migration adds any benefit here but I'm fine to do it.

Reply:
> The first line checks if the key exist or not so there won't be any exception.

You are missing the point, if one would downgrade for whatever reason, the version says the key exists, while in fact it doesn't.

Reply:
I have given it some more thought after and have changed it to make a migration hence why I marked this as solved.